### PR TITLE
updated object inspector settings

### PR DESCRIPTION
**Pull request by Barsonax**

### CODE REVIEW

File: Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs [hunk 1]
```diff
@@ -53,9 +54,36 @@ public bool EmptySelection
...
+			this.buttonSortByName.Checked = this.userSettings.SortByName;
+			this.Text = this.userSettings.TitleText;
+
+			if (this.userSettings.ExpandState != null)
+			{
+				this.gridExpandState.LoadFromXml(this.userSettings.ExpandState);
+			}
+		}
+
+		internal void SaveGridExpandState()
```
Comment (Barsonax) [inline Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs:77]: Not sure if this is right.

File: Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs [hunk 2]
```diff
@@ -53,9 +54,36 @@ public bool EmptySelection
 		public ObjectInspector(int runtimeId)
 		{
 			this.InitializeComponent();
+			this.buttonLock.CheckedChanged += (sender, args) => this.UserSettings.Locked = this.buttonLock.Checked;
```
Comment (Barsonax) [inline Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs:58]: The winforms designer is not working for me anymore so I added the changed handler myself here.

Not exactly sure why its not working anymore on visual studio 16.6.5.

File: Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs [hunk 3]
```diff
@@ -28,7 +23,13 @@ public partial class ObjectInspector : DockContent
...
 
 		private ExpandState              gridExpandState     = new ExpandState();
-
+		private ObjectInspectorState userSettings;
+		
+		public ObjectInspectorState UserSettings
+		{
+			get { return this.userSettings; }
+			set { this.userSettings = value; }
+		}
```
Comment (ilexp) [inline Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs:32]: Style: This property should follow the grouping of the others, i.e. not have a newline following it.

File: Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs [hunk 4]
```diff
@@ -28,7 +23,13 @@ public partial class ObjectInspector : DockContent
 		private ObjectSelection.Category displayCat          = ObjectSelection.Category.None;
 
 		private ExpandState              gridExpandState     = new ExpandState();
-
+		private ObjectInspectorState userSettings;
```
Comment (ilexp) [inline Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs:26]: Local style: Either remove the table-like alignment from all the fields, or make the new field align with the existing ones, as to not introduce inconsistency.

File: Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs (hunk 2 above)
Comment (ilexp) [reply Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs:58]: Since the designers failure is not "by design" (ha.. sorry 😅), but will likely be fixed in some future version, I'd prefer to put the event registration line manually into the proper spot inside `InitializeComponent` and make the registered method an actual named method in the class following the naming scheme. That way, the designer will pick it up without issues once it starts working again.

File: Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs (hunk 1 above)
Comment (ilexp) [reply Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs:77]: Can you go into detail why? I think I'm missing context.

File: Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorPlugin.cs [hunk 5]
```diff
@@ -30,7 +24,7 @@ internal static ObjectInspectorPlugin Instance
 
 		private	List<ObjectInspector>	objViews		= new List<ObjectInspector>();
 		private	bool					isLoading		= false;
-
+		private ObjectInspectorSettings userSettings;
```
Comment (ilexp) [inline Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorPlugin.cs:27]: Local style: Looks like the fields are meant to be laid out table-style, but the usage of inline tabs for alignment messes it up. Can you do a drive-by fix here and either remove the alignment from all the fields, or fix it by using inline-spaces? If you do the latter, make sure to also include the new field.

Comment (ilexp) [inline Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorPlugin.cs:27]: Style: Please do an explicit value assignment to `null` for consistency and readability reasons.

File: Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorPlugin.cs [hunk 6]
```diff
@@ -57,25 +51,24 @@ protected override IDockContent DeserializeDockContent(Type dockContentType)
...
-				XElement elem = new XElement("ObjectInspector");
-				elem.SetAttributeValue("id", i);
-				node.Add(elem);
-				this.objViews[i].SaveUserData(elem);
+				objectInspector.SaveGridExpandState();
 			}
+			pluginSettings.Set(this.userSettings);
 		}
-		protected override void LoadUserData(XElement node)
+
```
Comment (ilexp) [inline Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorPlugin.cs:62]: Style: Remove the newline here to keep the current member grouping in place, or (if changing the group layout) keep together `SaveUserData` and `LoadUserData` in the same group.

File: Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorSettings.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+namespace Duality.Editor.Plugins.ObjectInspector
+{
+	public class ObjectInspectorSettings
+	{
+		public List<ObjectInspectorState> ObjectInspectors { get; set; } = new List<ObjectInspectorState>();
+	}
+
+	public class ObjectInspectorState
+	{
+		private int id;
```
Comment (ilexp) [inline Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorSettings.cs:13]: Style: Please group all the fields and all the properties separately, as seen [here](https://github.com/Barsonax/duality/blob/ba2f86696a6dd28d92fb2d7b9061d1938091e96c/Source/Plugins/EditorModules/LogView/LogViewSettings.cs).

File: Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorSettings.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+using System.Xml.Linq;
+
+namespace Duality.Editor.Plugins.ObjectInspector
+{
+	public class ObjectInspectorSettings
+	{
+		public List<ObjectInspectorState> ObjectInspectors { get; set; } = new List<ObjectInspectorState>();
+	}
+
+	public class ObjectInspectorState
```
Comment (ilexp) [inline Source/Plugins/EditorModules/ObjectInspector/ObjectInspectorSettings.cs:11]: Class and public properties need XML docs! 🙂

**Review by ilexp — APPROVED** [review]
Implemented the requested changes myself due to the time constraints you mentioned on your side, and also fixed a bug in the multi-window scenario where IDs weren't yet used as needed to support this. Should be ready for merge now.
